### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Platon-jun**
- [DEMO LINK](https://platon-jun.github.io/layout_catalog/)
- [TEST REPORT LINK](https://platon-jun.github.io/layout_catalog/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -5,8 +5,430 @@
...
     <link rel="stylesheet" href="./styles/main.scss">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link
+      rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
   </head>
-  <body>
-    <h1>Catalog</h1>
+  <body class="body">
+    <header class="header header--margins">
```
Comment (vlad-demchuk) [inline src/index.html:14]:
```suggestion
    <header class="header">
```
do not set margins with the help of bem modifiers

just use mix for it (make this block bem element of `body` and apply external styles for it)

File: src/index.html [hunk 2]
```diff
@@ -5,8 +5,430 @@
...
+      <a class="header__img-link" href="#home">
+        <img
+          class="header__img"
+          src="images/logo.png"
+          alt="logo"
+        >
+      </a>
+
+      <nav class="header__nav">
```
Comment (vlad-demchuk) [inline src/index.html:23]: make it a separate bem block `nav`

File: src/index.html [hunk 3]
```diff
@@ -5,8 +5,430 @@
...
+          <li class="header__list">
+            <a class="header__link" href="#smartphones">Smartphones</a>
+          </li>
+
+          <li class="header__list">
+            <a
+              class="header__link"
+              data-qa="nav-hover"
+              href="#lap&Comp"
+            >Laptops & Computers</a>
```
Comment (vlad-demchuk) [inline src/index.html:42]: check your code style
```suggestion
            <a
              class="header__link"
              data-qa="nav-hover"
              href="#lap&Comp"
            >
              Laptops & Computers
            </a>
```

File: src/index.html [hunk 4]
```diff
@@ -5,8 +5,430 @@
...
+
+          <li class="header__list">
+            <a class="header__link" href="video">Video</a>
+          </li>
+        </ul>
+      </nav>
+    </header>
+    <main class="main">
+      <div class="product">
+        <div class="product-card" data-qa="card">
```
Comment (vlad-demchuk) [inline src/index.html:65]: it doesn't match bem naming convention

`product-card, product-card-image... and so on`

fix all such cases according to bem

File: src/index.html [hunk 5]
```diff
@@ -5,8 +5,430 @@
...
+            class="product-card-image"
+            src="./images/imac.jpeg"
+            alt="image apple A1219 iMac"
+          >
+          <h2 class="product-card-name">
+            APPLE A1419 iMac 27" Retina<br>
+            5K Monoblock (MNED2UA/A)
+          </h2>
+
+          <div class="product-card-container-code">
```
Comment (vlad-demchuk) [inline src/index.html:76]: redundant wrapper

File: src/index.html [hunk 6]
```diff
@@ -5,8 +5,430 @@
...
+              $2,199
+            </p>
+          </div>
+
+          <a
+            class="product-card-button"
+            href="/"
+            data-qa="card-hover"
+            > BUY
+          </a>
```
Comment (vlad-demchuk) [inline src/index.html:111]:
```suggestion
          <a
            class="product-card-button"
            href="/"
            data-qa="card-hover"
          > 
            BUY
          </a>
```

File: src/styles/blocks/header.scss [hunk 7]
```diff
@@ -0,0 +1,76 @@
+* {
+  margin: 0;
+  box-sizing: border-box;
+  font-family: Roboto, sans-serif;
```
Comment (vlad-demchuk) [inline src/styles/blocks/header.scss:4]: do not use `*` for `margins` or `font-family`

leave only `box-sizing` here

reset margin for some special tags, elements
move `font-family` to `html` selector

Comment (vlad-demchuk) [reply src/styles/blocks/header.scss:4]: move  it to a separate file like `reset.scss`

File: src/styles/blocks/header.scss [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    display: flex;
+  }
+
+  &__img {
+    height: 40px;
+    width: 40px;
+  }
+
+  &__nav {
+    display: flex;
```
Comment (vlad-demchuk) [inline src/styles/blocks/header.scss:25]: redundant `font-family`

File: src/styles/blocks/header.scss [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+    &-list {
+      font-size: $main-font-size;
+      font-weight: $font-weight-normal;
+      display: flex;
+      color: $main-text-color;
+      list-style: none;
+      padding: 0;
+      align-items: center;
```
Comment (vlad-demchuk) [inline src/styles/blocks/header.scss:34]: check css properties order
http://codeguide.academy/html-css.html#css-order

File: src/styles/blocks/header.scss [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  }
+
+  &__list {
+    margin-right: 20px;
+
+    &:last-child {
```
Comment (vlad-demchuk) [inline src/styles/blocks/header.scss:41]: you can use `:not(:last-child)`

File: src/styles/blocks/header.scss [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  &__link {
+    display: block;
+    text-transform: uppercase;
+    color: inherit;
+    font-size: inherit;
+    text-decoration: none;
+    padding-top: 23px;
+    padding-bottom: 23px;
```
Comment (vlad-demchuk) [inline src/styles/blocks/header.scss:53]: use padding shorthand property

**Review by vlad-demchuk — CHANGES_REQUESTED** [review]
nothing should be broken when I add one more card
image

**Review by vlad-demchuk — CHANGES_REQUESTED** [review]

**Review by nazarmatsevych — CHANGES_REQUESTED** [review]
1. According to the task description
image

So you need to use width instead of max-width for your card container

**Pull request closed without merging**